src/services/api.service.js

```diff
@@ -12,19 +13,162 @@ export function get (endpoint) {
     });
 }
 
-export function post (endpoint, payload = {}) {
+export function getFavoriteBooks () {
     return new Promise((resolve) => {
-        fetch(`../data/${endpoint}.json`, {
-            method: 'POST',
-            headers: {
-                'Content-Type': 'application/json'
-            },
-            body: JSON.stringify(payload),
-        })
-            .then(() => {
+        fetch(`../data/books.json`)
+            .then((response) => response.json())
+            .then((json) => {
+                setTimeout(() => {
+                    resolve(json.data.filter((book) => book.favorite));
+                }, serverLatency);
+            });
+    });
+}
+
+export function getBookByIsbn (isbn) {
+    return new Promise((resolve, reject) => {
+        fetch(`../data/books.json`)
+            .then((response) => response.json())
+            .then((json) => {
                 setTimeout(() => {
-                    resolve();
+                    const book = json.data.find((book) => book.isbn === isbn);
+                    if (!book) {
+                        reject(new Error('Book not found'));
+                    }
+                    resolve(book);
                 }, serverLatency);
             });
     });
 }
+
+export function createBook (book) {
+    return new Promise((resolve, reject) => {
+        getBookByIsbn(book.isbn)
+            .catch(() => {
+                validateBook(book).catch((error) => reject(error));
+
+                const defaultValues = {
+                    pageCount: -1,
+                    publishedDate: '',
+                    thumbnailUrl: 'https://picsum.photos/150/190',
+                    shortDescription: '',
+                    longDescription: '',
+                    categories: [],
+                    favorite: false,
+                };
+                const bookToCreate = {
```

Comment:
do we want to add in an `id` field? I know we have `isbn` but traditionally we would add an `id`, use something like `uuid` ?

Reply:
I did actually first have the function create a unique id for each book, but then removed it seemed redundant and made the database simpler and maybe easier to understand?

I think for this workshop it's probably best if they don't worry much about the backend because
- introducing those concepts on top of everything else is a lot
- if we were going to introduce those topics I think it would probably be better done with an actual backend, because this isn't the most accurate representation anyway and so might just be confusing. I think learning backend would be better done when you're doing it properly and then you also have online guides and resources to help rather than trying to do it with this fake version.